packages/muelu/cmake/MueLuInstall.cmake

```diff
@@ -0,0 +1,59 @@
+# Install an executable and dependent files to the same directory as
+# ${CMAKE_CURRENT_SOURCE_DIR}, but under ${CMAKE_INSTALL_PREFIX}.
+#
+# For example, if called in packages/muelu/example/basic/CMakeLists.txt, then we will install to
+#   ${CMAKE_INSTALL_PREFIX}/packages/muelu/example/basic/
+#
+# Usage::
+#
+#   muelu_install( <exe_root_name>
+#     [ADDITIONAL_FILES <file0> <file1> ...]
+#     )
+#
+# For example, we might have added an executable and a couple of input files like so:
+#
+# tribits_add_executable(
+#      Driver
+#      SOURCES driver_source.cpp
+# )
+#
+# set(XML_INPUTS
+#    mediocre_input.xml amazing_input.xml
+# )
+#
+# tribits_copy_files_to_binary_dir(
+#     inputs_cp
+#     SOURCE_FILES
+#     ${XML_INPUTS}
+# )
+#
+# In order to add these to the install, we do
+#
+# muelu_install(
+#    Driver
+#    ADDITIONAL_FILES
+#    ${XML_INPUTS}
+#  )
+#
+function(muelu_install TARGET)
+
+  cmake_parse_arguments(
+    PARSE
+    ""
+    ""
+    "ADDITIONAL_FILES"
+    ${ARGN}
+  )
+
+  # set the install directory to the current source folder, but relative to install prefix
+  set(INSTALL_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+  file(RELATIVE_PATH INSTALL_DIRECTORY ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+  set(INSTALL_DIRECTORY ${CMAKE_INSTALL_PREFIX}/${INSTALL_DIRECTORY})
```

Comment:
So this would install under:

```
<prefix>/packages/muelu/example/basic/
```

for example, in the call from:

```
<trilinosSrcDir>/packages/muelu/example/basic/CMakeLists.txt
```

?

Would in not be better to install these under:

```
<prefix>/examples/muelu/basic/
```

?

That would make it clear that the user will only find examples under `<prefix>/examples/`.  Otherwise, they might thing they will find packages under `<prefix>/packages/` (for whatever "packages" might mean to them).

This is just my opinion but it does impact the overall understandability of the Trilinos install tree so this should be reviewed by the Framework team and other product area leads as well.

Reply:
@bartlettroscoe I don't think I really care where it gets installed, as long as there is some sensible pattern to it. I like the current approach, as it makes it easy to figure out where the corresponding source lives.

Reply:
Right, but the base level install directory structure should be agreed upon by Trilinos as a whole.  Just having a `<prefix>/packagess/` directory show up should be decided upon at the Trilinos level.  And there is also a common-look and feel issue if package's don't do things in a consistent way.

Reply:
Yep, fully agree.